SmartDeviceLink/SDLChoiceSetManager.m

```diff
@@ -459,18 +513,79 @@ - (SDLKeyboardProperties *)sdl_defaultKeyboardConfiguration {
 
 #pragma mark - Getters
 
+- (NSMutableSet<SDLChoiceCell *> *)preloadedMutableChoices {
+    __block NSMutableSet<SDLChoiceCell *> *set = nil;
+    [self sdl_runSyncOnQueue:^{
+        set = self->_preloadedMutableChoices;
+    }];
+
+    return set;
+}
+
+- (NSMutableSet<SDLChoiceCell *> *)pendingMutablePreloadChoices {
+    __block NSMutableSet<SDLChoiceCell *> *set = nil;
+    [self sdl_runSyncOnQueue:^{
+        set = self->_pendingMutablePreloadChoices;
+    }];
+
+    return set;
+}
+
 - (NSSet<SDLChoiceCell *> *)preloadedChoices {
-    return [_preloadedMutableChoices copy];
+    __block NSSet<SDLChoiceCell *> *set = nil;
+    [self sdl_runSyncOnQueue:^{
+        set = [self->_preloadedMutableChoices copy];
+    }];
+
+    return set;
 }
 
 - (NSSet<SDLChoiceCell *> *)pendingPreloadChoices {
-    return [_pendingMutablePreloadChoices copy];
+    __block NSSet<SDLChoiceCell *> *set = nil;
+    [self sdl_runSyncOnQueue:^{
+        set = [self->_pendingMutablePreloadChoices copy];
+    }];
+
+    return set;
+}
```

Comment:
Same as above

Reply:
Since the `pendingMutablePreloadChoices` synchronized getter was removed, this is still needed. My understanding is that a crash will happen if a thread attempts to read while another thread is writing, then a crash can occur.